src/librbd/internal.cc

```diff
@@ -256,6 +256,7 @@ int validate_pool(IoCtx &io_ctx, CephContext *cct) {
   if (r < 0) {
     lderr(cct) << "failed to release self-managed snapshot " << snap_id
                << ": " << cpp_strerror(r) << dendl;
+    return r;
```

Comment:
Not returning the error result was intentional.  There is no way to recover and delete the snapshot id ever again -- and the pool has already been changed to self-managed snapshot mode.  Therefore, no need to abort the create request.